### PR TITLE
added page about newlines and `format_board` to Ch 11 Tic-Tac-Toe #3

### PR DESCRIPTION
**Pull request by spamegg1**
Happy New Year! #3

(the "next page" link right before bonus challenge is missing for now, because the next page doesn't exist yet)

### CODE REVIEW

File: backend/main/chapters/c11_tic_tac_toe_project.py [hunk 1]
```diff
@@ -589,3 +589,346 @@ def generate_inputs(cls):
...
+    print_board([
+        ['X', 'O', 'X'],
+        [' ', 'O', 'O'],
+        [' ', 'X', ' ']
+    ])
+
+(What's `"".join`? Google it!)
+        """
+
+        program_in_text = False
```
Comment (alexmojaki) [inline backend/main/chapters/c11_tic_tac_toe_project.py:615]: why?

Comment (spamegg1) [reply backend/main/chapters/c11_tic_tac_toe_project.py:615]: Hmm... I think I was under the false impression that `__program_indented__` has to be the very last thing in the class string.

Comment (spamegg1) [reply backend/main/chapters/c11_tic_tac_toe_project.py]: Should be fixed in the next commit.

**Comment by spamegg1** [comment]
The only way I could find to display
```python
len('\n')
```
correctly was
```
`__program__`
```
Tried other things like
```
__program__
__program_indented__
`__program_indented__`
```
, I would get either
```
len('

')
```
or
```
len(' ')
```
It still doesn't get syntax coloring though. 
Also if the user copies it from the text and pastes into the shell, it looks correct but does not pass the Step because it's actually `len('\\n')` that is copied even though it looks `len('\n')`.
Can't figure this out. It's probably something stupidly simple.

**Comment by alexmojaki** [comment]
Apparently Python is weird about slashes in replacement strings when using `re`:

```python
>>> "hello bob".replace("bob", "\\n")
'hello \\n'
>>> re.sub("bob", "\\n", "hello bob")
'hello \n'
>>> re.sub("bob", "\\\\n", "hello bob")
'hello \\n'
```

The problem line was this:

```python
text = re.sub(r" *__program_indented__", indented, text, flags=re.MULTILINE)
```

**Comment by alexmojaki** [comment]
I forgot that the bonus format_board would require converting int to string. Let's save that as an exercise for after they learn about types.

In the meantime I suggest changing the bonus challenge to a prettier board format:

```python
def format_board(board):
    joined_rows = []
    for row in board:
        joined_rows.append("|".join(row))
    lines = []
    for _ in board[0]:
        lines.append("-")
    line = f'\n{"+".join(lines)}\n'
    return line.join(joined_rows)

print(format_board(board = [
    ['X', 'O', 'X'],
    [' ', 'O', 'O'],
    [' ', 'X', ' ']
]))
```

Which prints

```
X|O|X
-+-+-
 |O|O
-+-+-
 |X| 
```

**Comment by alexmojaki** [comment]
Merge if you're happy!

**Comment by spamegg1** [comment]
Looks pretty good... I didn't think to add `TODO` for the link, good idea.
Merging...

**Pull request merged**